support building docs without API part

With the changes in #79, having `easybuild-framework` cloned into `src` is a hard requirement for the docs build to work:

```
$ ls src
ls: src: No such file or directory
$ make preview
mkdocs serve
INFO     -  Building documentation...
INFO     -  Cleaning site directory
FileNotFoundError: [Errno 2] No such file or directory: 'py3.py' -> 'src/easybuild/tools/py2vs3/py2.py'
make: *** [preview] Error 1
```

That's because the `gen_ref_pages.py` script blindly assumes that `src` contains a copy of the `easybuild-framework` repository.

We should allow building the documentation without the API part, to speed up local development of the documentation.

While doing that, we should also use `src/easybuild-framework` instead of just `src`.

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -22,7 +22,7 @@ jobs:
         uses: actions/checkout@v3
         with:
           repository: easybuilders/easybuild-framework
-          path: src
+          path: src/easybuild-framework
 
       - name: install mkdocs
         run: |

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -33,7 +33,7 @@ jobs:
         uses: actions/checkout@v3
         with:
           repository: easybuilders/easybuild-framework
-          path: src
+          path: src/easybuild-framework
 
       - name: install mkdocs
         run: |

--- a/docs/gen_ref_pages.py
+++ b/docs/gen_ref_pages.py
@@ -8,22 +8,27 @@ from pathlib import Path
 import mkdocs_gen_files
 
 
-if not Path('src/easybuild-framework/easybuild').is_dir():
+EB_FRAMEWORK = "src/easybuild-framework"
+
+if not Path(f"{EB_FRAMEWORK}/easybuild").is_dir():
     print("WARNING  -  EasyBuild Framework is not available. API documentation will not be generated.")
     exit(0)
 
 
-# remove the problematic py2.py and symlink the py3.py in its place
-Path("src/easybuild-framework/easybuild/tools/py2vs3/py2.py").unlink(missing_ok=True)
-Path("src/easybuild-framework/easybuild/tools/py2vs3/py2.py").symlink_to(Path("py3.py"))
+try:
+    # remove the problematic py2.py and symlink the py3.py in its place
+    Path(f"{EB_FRAMEWORK}/easybuild/tools/py2vs3/py2.py").unlink(missing_ok=True)
+    Path(f"{EB_FRAMEWORK}/easybuild/tools/py2vs3/py2.py").symlink_to(Path("py3.py"))
+except FileNotFoundError:
+    print("WARNING  -  Unable to create py2.py -> py3.py symlink")
 
 # build a navigation for the menu and a dictionary of navigations for each section
 menu_nav = mkdocs_gen_files.Nav()
 navs = {}
 
-for path in sorted(Path("src/easybuild-framework/easybuild/").rglob("*.py")):
-    module_path = path.relative_to("src/easybuild-framework").with_suffix("")
-    doc_path = path.relative_to("src/easybuild-framework").with_suffix(".md")
+for path in sorted(Path(f"{EB_FRAMEWORK}/easybuild/").rglob("*.py")):
+    module_path = path.relative_to(EB_FRAMEWORK).with_suffix("")
+    doc_path = path.relative_to(EB_FRAMEWORK).with_suffix(".md")
     full_doc_path = Path("api", doc_path)
 
     parts = list(module_path.parts)

--- a/docs/gen_ref_pages.py
+++ b/docs/gen_ref_pages.py
@@ -8,17 +8,22 @@ from pathlib import Path
 import mkdocs_gen_files
 
 
+if not Path('src/easybuild-framework/easybuild').is_dir():
+    print("WARNING  -  EasyBuild Framework is not available. API documentation will not be generated.")
+    exit(0)
+
+
 # remove the problematic py2.py and symlink the py3.py in its place
-Path("src/easybuild/tools/py2vs3/py2.py").unlink(missing_ok=True)
-Path("src/easybuild/tools/py2vs3/py2.py").symlink_to(Path("py3.py"))
+Path("src/easybuild-framework/easybuild/tools/py2vs3/py2.py").unlink(missing_ok=True)
+Path("src/easybuild-framework/easybuild/tools/py2vs3/py2.py").symlink_to(Path("py3.py"))
 
 # build a navigation for the menu and a dictionary of navigations for each section
 menu_nav = mkdocs_gen_files.Nav()
 navs = {}
 
-for path in sorted(Path("src/easybuild/").rglob("*.py")):
-    module_path = path.relative_to("src").with_suffix("")
-    doc_path = path.relative_to("src").with_suffix(".md")
+for path in sorted(Path("src/easybuild-framework/easybuild/").rglob("*.py")):
+    module_path = path.relative_to("src/easybuild-framework").with_suffix("")
+    doc_path = path.relative_to("src/easybuild-framework").with_suffix(".md")
     full_doc_path = Path("api", doc_path)
 
     parts = list(module_path.parts)

--- a/mkdocs.yml
+++ b/mkdocs.yml
@@ -66,7 +66,7 @@ plugins:
       default_handler: python
       handlers:
         python:
-          paths: [src]
+          paths: [src/easybuild-framework]
           options:
             docstring_style: sphinx
             docstring_section_style: spacy

